### PR TITLE
refactor: General cleanup

### PR DESCRIPTION
**Pull request by alexanderroidl**
- Cleaned up and standardised CAPTCHA parameter's type definitions
- Moved main "Solver" class to it's own file
- Fixed misspellings in code, as well as in the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 <b>[English](README.md)</b> (recommended) | [中国语文科](README.zh.md) | [Русский](README.ru.md)
 
 
-# JavaScript captcha solver module for 2Captcha API
```
Comment (alexanderroidl) [inline README.md:4]: This package does not specifically concern JS only anymore, which is why I changed this.

File: src/structs/2captcha.ts [hunk 2]
```diff
@@ -1,18 +1,4 @@
...
-import { APIError } from "./2captchaError"
-import * as utils from "../utils/generic"
-import  getProviderData  from "./providers/providers"
-import { softId } from "./constants/constants"
-import checkCaptchaParams from "../utils/checkCaptchaParams"
-
-
-const provider = getProviderData ()
-
-interface BaseSolve {
```
Comment (alexanderroidl) [inline src/structs/2captcha.ts:11]: This has no effect whatsoever at runtime as it's empty and it's generally unused, which is why I removed it.

File: src/structs/2captcha.ts [hunk 3]
```diff
@@ -108,19 +75,31 @@ export interface yandexSmart {
...
+    proxy?: string,
+    proxytype?: string,
+    userAgent?: string
+}
+
 /**
- * Interface for GeeTest V4 captcha
+ * This concerns GeeTest CAPTCHA's version 4
  * 
- * @typedef {object} paramsGeeTestV4
```
Comment (alexanderroidl) [inline src/structs/2captcha.ts:114]: @typedef annotations aren't necessary in TypeScript (only for JS).

File: src/structs/2captcha.ts (hunk 2 above)
Comment (dzmitry-duboyski) [reply src/structs/2captcha.ts:11]: Hi. Yes indeed the BaseSolve interface does not make sense. We will remove it in the next release.

**Comment by dzmitry-duboyski** [comment]
@alexanderroidl Your suggestions have been partially applied in this [PR](https://github.com/2captcha/2captcha-javascript/pull/6). Thank you for describing these problems.

**Pull request closed without merging**